src/modules/app/components/Navigator.js

```diff
@@ -74,12 +79,86 @@ export const AppStack = createStackNavigator(
   }
 )
 
-export const LandingStack = createSwitchNavigator(
-  {
-    'Landing': LandingContainer,
-    'App': AppStack
-  }
-)
+export const LandingStack = createSwitchNavigator({
+  // TODO replace this ActivityIndicator, NATIVE-211
+  'Loading': ActivityIndicator,
+  'Landing': LandingContainer,
+  'App': AppStack
+})
 
 const AppContainer: NavigationContainer<NavigationState, {}, {}> = createAppContainer(LandingStack)
-export default AppContainer
+
+type PropsType = {|
+  fetchCategory: (cityCode: string, language: string, key: string) => void,
+  clearCategory: (key: string) => void,
+  fetchCities: () => void
+|}
+
+class Navigator extends React.Component<PropsType> {
+  localizationSettings: LocalizationSettings
+  navigator: {current: null | React$ElementRef<NavigationContainer<NavigationState, {}, {}>>}
+
+  constructor (props: PropsType) {
+    super(props)
+    this.localizationSettings = new LocalizationSettings()
+    this.navigator = React.createRef()
```

Comment:
I'm pretty sure we can move the `navigator` reference to a child of this component.

Can the function `loadSelectedCity` be called in the `componentWillMount` of a child component? Then you can use the `withNavigation` hoc to get a reference to the navigator

Reply:
Could you elaborate further what you mean by that? If I understand you correctly, you propose to introduce a child component to handle the navigating as soon as it is known whether a city is selected or not.
Afaics you can't do any navigating outside the `AppContainer` without using a `ref`. I think `withNavigation` would only work inside a child component of the `AppContainer`, since the `navigation` is only defined there.

Reply:
Yes that's what I meant. Not sure whether that works in this case though. The idea would be, as you said, to create a child component.

Afaik refs shouldn't be needed in 99% of the cases.